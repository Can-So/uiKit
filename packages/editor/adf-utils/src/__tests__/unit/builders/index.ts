import {
  doc,
  p,
  emoji,
  a,
  em,
  b,
  u,
  mention,
  breakout,
  codeBlock,
  text,
} from '../../../builders';

describe('Builders', () => {
  const nodes = [
    () =>
      doc(
        p(
          'My favourite emoji is ',
          emoji({ text: '🤦‍♂️', shortName: ':man_facepalming:' }),
          ' . What is yours?',
        ),
      ),
    () =>
      doc(
        p(
          a({ href: 'https://www.atlassian.com' })('Hello, World!'),
          ' Look I can do',
          em('italic'),
          em(b(', strong ')),
          em(b(u('and underlined text!'))),
          ' and action mark and invalid action mark.',
        ),
      ),
    () =>
      doc(
        p(
          'Hi, my name is... My name is... My name is... My name is ',
          mention({ id: '1', text: '@Oscar Wallhult' }),
          ' :D',
        ),
        p(
          'This is a ',
          mention({ id: '2', text: '@mention', accessLevel: '' }),
          '. And this is a broken ',
          mention({ id: 'mention', text: '@unknown' }),
        ),
        p(
          'Mentions with generic ids ',
          mention({ id: 'here', text: '@here', accessLevel: 'CONTAINER' }),
          ' ',
          mention({ id: 'all', text: '@all', accessLevel: 'CONTAINER' }),
        ),
      ),
    () =>
      doc(
        p(emoji({ shortName: ':grinning:', text: '😀' }), ' '),
        p(emoji({ shortName: ':man_facepalming:', text: '🤦‍♂️' }), ' '),
        p(emoji({ shortName: ':flag_ru:', text: '🇷🇺' }), ' '),
        p(emoji({ shortName: ':wtf:', text: ':wtf:' }), ' '),
      ),
    () => doc(breakout({ mode: 'wide' })(codeBlock({})(text('some code')))),
  ];

  nodes.forEach((node, idx) => {
    it(`should be able to generate correct ADF for example #${idx}`, () => {
      expect(node()).toMatchSnapshot();
    });
  });
});
