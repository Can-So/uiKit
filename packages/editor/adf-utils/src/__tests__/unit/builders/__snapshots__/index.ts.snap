// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Builders should be able to generate correct ADF for example #0 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "My favourite emoji is ",
          "type": "text",
        },
        Object {
          "attrs": Object {
            "shortName": ":man_facepalming:",
            "text": "ü§¶‚Äç‚ôÇÔ∏è",
          },
          "type": "emoji",
        },
        Object {
          "text": " . What is yours?",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Builders should be able to generate correct ADF for example #1 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "href": "https://www.atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "Hello, World!",
          "type": "text",
        },
        Object {
          "text": " Look I can do",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "em",
            },
          ],
          "text": "italic",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "strong",
            },
            Object {
              "type": "em",
            },
          ],
          "text": ", strong ",
          "type": "text",
        },
        Object {
          "marks": Array [
            Object {
              "type": "underline",
            },
            Object {
              "type": "strong",
            },
            Object {
              "type": "em",
            },
          ],
          "text": "and underlined text!",
          "type": "text",
        },
        Object {
          "text": " and action mark and invalid action mark.",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Builders should be able to generate correct ADF for example #2 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Hi, my name is... My name is... My name is... My name is ",
          "type": "text",
        },
        Object {
          "attrs": Object {
            "accessLevel": "",
            "id": "1",
            "text": "@Oscar Wallhult",
          },
          "type": "mention",
        },
        Object {
          "text": " :D",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "text": "This is a ",
          "type": "text",
        },
        Object {
          "attrs": Object {
            "accessLevel": "",
            "id": "2",
            "text": "@mention",
          },
          "type": "mention",
        },
        Object {
          "text": ". And this is a broken ",
          "type": "text",
        },
        Object {
          "attrs": Object {
            "accessLevel": "",
            "id": "mention",
            "text": "@unknown",
          },
          "type": "mention",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "text": "Mentions with generic ids ",
          "type": "text",
        },
        Object {
          "attrs": Object {
            "accessLevel": "CONTAINER",
            "id": "here",
            "text": "@here",
          },
          "type": "mention",
        },
        Object {
          "text": " ",
          "type": "text",
        },
        Object {
          "attrs": Object {
            "accessLevel": "CONTAINER",
            "id": "all",
            "text": "@all",
          },
          "type": "mention",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Builders should be able to generate correct ADF for example #3 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "shortName": ":grinning:",
            "text": "üòÄ",
          },
          "type": "emoji",
        },
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "shortName": ":man_facepalming:",
            "text": "ü§¶‚Äç‚ôÇÔ∏è",
          },
          "type": "emoji",
        },
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "shortName": ":flag_ru:",
            "text": "üá∑üá∫",
          },
          "type": "emoji",
        },
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "shortName": ":wtf:",
            "text": ":wtf:",
          },
          "type": "emoji",
        },
        Object {
          "text": " ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`Builders should be able to generate correct ADF for example #4 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {},
      "content": Array [
        Object {
          "text": "some code",
          "type": "text",
        },
      ],
      "marks": Array [
        Object {
          "attrs": Object {
            "mode": "wide",
          },
          "type": "breakout",
        },
      ],
      "type": "codeBlock",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
